pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -402,6 +402,19 @@ func handleModify(ctx *downloaderContext, key string,
 	if config.RefCount != 0 && (status.HasError() || status.State != types.DOWNLOADED) {
 		log.Functionf("handleModify installing %s", config.Name)
 		handleCreate(ctx, config, status, key, receiveChan)
+		// Retrying the download due to image verification failure
+		// This happens when: RefCount and retryCount is non-zero,
+		// and DownloaderStatus state is "downloaded"
+	} else if config.RefCount != 0 && config.BlobDownloadRetryCount != 0 &&
```

Comment:
Are we sure that we can not match the if on line 402 when we intent to do a retry? 
I think it would be safer to put this as the first if statement, and have line 402 in an else if statement.

Reply:
I have reverted the "else if' as it was before.